nodebuilder/state/state.go

```diff
@@ -79,47 +81,154 @@ type Module interface {
 // API is a wrapper around Module for the RPC.
 // TODO(@distractedm1nd): These structs need to be autogenerated.
 type API struct {
-	IsStopped         func() bool
-	Balance           func(ctx context.Context) (*state.Balance, error)
-	BalanceForAddress func(ctx context.Context, addr state.Address) (*state.Balance, error)
-	Transfer          func(
-		ctx context.Context,
-		to state.AccAddress,
-		amount math.Int,
-		gasLimit uint64,
-	) (*state.TxResponse, error)
-	SubmitTx         func(ctx context.Context, tx state.Tx) (*state.TxResponse, error)
-	SubmitPayForData func(ctx context.Context, nID namespace.ID, data []byte, gasLim uint64) (
-		*state.TxResponse,
-		error,
-	)
-	CancelUnbondingDelegation func(
-		ctx context.Context,
-		valAddr state.ValAddress,
-		amount,
-		height state.Int,
-		gasLim uint64,
-	) (*state.TxResponse, error)
-	BeginRedelegate func(
-		ctx context.Context,
-		srcValAddr,
-		dstValAddr state.ValAddress,
-		amount state.Int,
-		gasLim uint64,
-	) (*state.TxResponse, error)
-	Undelegate func(ctx context.Context, delAddr state.ValAddress, amount state.Int, gasLim uint64) (
-		*state.TxResponse,
-		error,
-	)
-	Delegate func(ctx context.Context, delAddr state.ValAddress, amount state.Int, gasLim uint64) (
-		*state.TxResponse,
-		error,
-	)
-	QueryDelegation    func(ctx context.Context, valAddr state.ValAddress) (*types.QueryDelegationResponse, error)
-	QueryUnbonding     func(ctx context.Context, valAddr state.ValAddress) (*types.QueryUnbondingDelegationResponse, error)
-	QueryRedelegations func(
-		ctx context.Context,
-		srcValAddr,
-		dstValAddr state.ValAddress,
-	) (*types.QueryRedelegationsResponse, error)
+	Internal struct {
+		AccountAddress    func(ctx context.Context) (state.Address, error)                      `perm:"read"`
```

Comment:
Hmm, I don't think the address of the node should be public

Reply:
good catch

Reply:
done